docs/_extensions/automembersummary.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2014-2015 Canonical Limited.
+#
+# This file is part of charm-helpers.
+#
+# charm-helpers is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# charm-helpers is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with charm-helpers.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import inspect
+
+from docutils.parsers.rst import directives
+from sphinx.ext.autosummary import Autosummary
+from sphinx.ext.autosummary import get_import_prefixes_from_env
+from sphinx.ext.autosummary import import_by_name
+
+
+class AutoMemberSummary(Autosummary):
```

Comment:
This has been cargo-culted into a few repos now.  I really should extract this and put it on pypi.